data/dataset.py

```diff
@@ -0,0 +1,97 @@
+import random
+
+import torch
+import numpy as np
+from torch.utils.data import Dataset
+from datasets import Dataset as HFDataset
+
+from data.masking import AwesomeMasks
+from data.tokenizer import MidiEncoder
+from data.quantizer import MidiQuantizer
+from data.augmentation import pitch_shift, change_speed
+
+
+class MidiDataset(Dataset):
+    def __init__(
+        self, dataset: HFDataset, tokenizer: MidiEncoder, augmentation_probability: float = 0.0, masking_probability: float = 0.15
+    ):
+        super().__init__()
+
+        self.dataset = dataset
+        self.quantizer = MidiQuantizer(7, 7, 7)
+        self.tokenizer = tokenizer
+        self.masks = AwesomeMasks()
+
+        self.augmentation_probability = augmentation_probability
+        self.masking_probability = masking_probability
+
+    def __len__(self):
+        return len(self.dataset)
+
+    def apply_augmentation(self, record: dict):
+        # shift pitch augmentation
+        if random.random() < self.augmentation_probability:
+            shift = 7
+            record["pitch"] = pitch_shift(record["pitch"], shift)
+
+        # change tempo augmentation
+        if random.random() < self.augmentation_probability:
+            record["dstart"], record["duration"] = change_speed(record["dstart"], record["duration"])
+            # change bins for new dstart and duration values
+            record["dstart_bin"] = np.digitize(record["dstart"], self.quantizer.dstart_bin_edges) - 1
+            record["duration_bin"] = np.digitize(record["duration"], self.quantizer.duration_bin_edges) - 1
```

Comment:
```suggestion
        if random.random() < self.pitch_shift_probability:
            shift = 7
            record["pitch"] = pitch_shift(record["pitch"], shift)

        # change tempo augmentation
        if random.random() < self.time_stretch_probability:
            record["dstart"], record["duration"] = change_speed(record["dstart"], record["duration"])
            # change bins for new dstart and duration values
            record["dstart_bin"] = np.digitize(record["dstart"], self.quantizer.dstart_bin_edges) - 1
            record["duration_bin"] = np.digitize(record["duration"], self.quantizer.duration_bin_edges) - 1
```

If you use the same value twice, it looses it's "probability" meaning. Let's have more precise control over the augmentation